core/modules/node/node.block.inc

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ *  NodeBlock  extends Block
+ *  
+ *  This class allows us to display a particular node in a block.
+ */
+class NodeBlock extends Block {
+  /**
+   *  Sets title text on draggable block panel in Layout builder.
+   */
+  function getAdminTitle() {
+    return t('Content block');
+  }
+
+  /**
+   * Return a preview for this block.
+   */
+  function getAdminPreview() {
+    return t('Displays content of node ID: @nid', array('@nid' => $this->settings['nid']));
+  }
+
+  /**
+   *  Sets block subject on block view.
+   */
+  function getTitle() {
+    $title = NULL;
+    $node = node_load($this->settings['nid']);
+    if ($this->settings['title_display'] === LAYOUT_TITLE_CUSTOM && $this->settings['title']) {
+      $title = check_plain($this->settings['title']);
+    }
+    elseif ($this->settings['title_display'] === LAYOUT_TITLE_DEFAULT) {
+      $title = $node->title;
+    }
+
+    if ($title && $this->settings['link_node_title']) {
+      $title = l($title, 'node/' . $node->nid);
+    }
+
+    return $title;
+  }
+
+  /**
+   *  Sets block content on block view.
+   */
+  function getContent() {
+    $node = node_load($this->settings['nid']);
+
+    // Prevent display of no access to the node.
+    if (!node_access('view', $node)) {
+      return;
+    }
+
+    // Use a clone of the node so we can hide the title safely.
+    $clone = clone($node);
+
+    // Hide node title if not required.
+    if (empty($this->settings['leave_node_title'])) {
+      $clone->title = '';
+    }
+
+    $content = node_view($clone, $this->settings['view_mode']);
+
+    // Hide links if they've been suppressed.
+    if (empty($this->settings['links'])) {
+      $content['links']['#access'] = FALSE;
+    }
+
+    // Add 'submitted by' settings.
+    if (!empty($this->settings['display_submitted'])) {
+      $content['#display_submitted'] = TRUE;
+    }
+
+    // Add theme hook suggestion.
+    $content['#theme'] = 'node__content-block-' . $this->settings['nid'];
```

Comment:
I don't think theme suggestions should include hyphens. Any template name is converted to hyphens automatically.

Here are the current suggestions for nodes:
```
  $variables['theme_hook_suggestions'][] = 'node__' . $node->type;
  $variables['theme_hook_suggestions'][] = 'node__' . $node->nid;
```

I think these suggestions will continue to work. So perhaps we could simplify this suggestion down into just `node__block`? Though you would get a namespace conflict if you created a node type "block" :stuck_out_tongue: